cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -191,13 +274,38 @@ class ScalarAggregateNode : public ExecNode, public TracedNode {
 
     const auto& aggregate_options = checked_cast<const AggregateNodeOptions&>(options);
     auto aggregates = aggregate_options.aggregates;
+    const auto& keys = aggregate_options.keys;
+    const auto& segment_keys = aggregate_options.segment_keys;
+
+    if (keys.size() > 0) {
```

Comment:
This could probably be a DCHECK.  I don't think it should be possible to get here given that the factory in RegisterAggregateNode should check for this.

Reply:
Because a future maintainer may well call this function directly or expose it, I think this check can stay at least for safety purposes.